### ISSUE TITLE
MaxVolumeCount

### ISSUE THREAD

**Issue by DanRStevens** [OutpostUniverse/op2ext#20]
In #16 we briefly discussed adding a `MaxVolumeCount` field relating to the `VolList` class. I'm opening this to track further discussion.

The `MaxVolumeCount` field would be a named constant to specify how many VOL files the VolList class was configured to allow loading, which is set as a compile time constant.

There was some concern expressed about adding to the interface for op2ext. I'd say we should keep the new constant out of the main interface. This constant should probably go in VolList.h, not op2ext.h. In particular, op2ext.h represents the public interface to op2ext, and as it does not include any other project header files, it is the sole source of information for the public interface. Anything added to VolList.h would not become part of the public interface.

The constant would be `public` in terms of the C++ language rules for a compilation unit that includes VolList.h. It would not be public in terms of the published API spec. Additionally, the value would not be exported, and so also not part of the ABI. I suppose you could claim it relates to the ABI, as constant values in headers effectively become baked into the compiled code. That means if the header was imported by both op2ext, and a module that depends on op2ext, they may see two different values for the constant if they were compiled from different versions of the header file. That could result in incompatibilities. However, VolList.h is not a public header. It should not be included from external modules.

It does however seem permissible for VolList.h to be included by a test project bundled with op2ext. In that sense, test code is already strongly dependent on matching versions. It's also generally appropriate to give test code a little more access then client code would generally need. As such, I think providing access to this constant as a public member of a non public header file is an acceptable compromise.

**Comment by Brett208** [comment]
I'm going to close this since the const variable now exists and is called by the TestModule. While this was a small change, it uncovered what was wrong in the TestModule for testing attempting to load too many volumes, so glad we took the time to implement it.
